totalRP3/Core/StringUtil.lua

```diff
@@ -27,12 +27,24 @@ function TRP3_StringUtil.SortCompareStrings(a, b)
 	return strcmputf8i(a, b) < 0;
 end
 
-local function CanonicalizeSearchString(str)
-	str = string.gsub(str, "[%p%c%s]", "-");
-	str = string.utf8lower(str);
+local function GenerateSearchableString(str)
+	str = string.gsub(str, "[%p%c%s]+", "");
+	str = string.lower(str);
```

Comment:
To also note; at some point in recent client history string.lower became UTF-8 aware. We don't know when this happened, but it means that we can stop using the slow-as-balls `string.utf8lower` library function generally.